src/pocketmine/Player.php

```diff
@@ -2864,6 +2864,10 @@ public function handleDropItem(DropItemPacket $packet) : bool{
 			// Windows 10 Edition drops the contents of the crafting grid on container close - including air.
 			return true;
 		}
+		
+		if($this->gamemode === self::SPECTATOR) {
```

Comment:
It's a micro optimization yes, but what if a server has 9000 players (on 1 PM server not realistic but theoretical) and the function is run for each of them? Every micro optimization counts ;)